### PR TITLE
[GE] Color Picker

### PR DESCRIPTION
**Pull request by msDestiny14**
Color picker now for the GUI Editor:
The code has been added to the shared Component for Color3LineComponent. However it is not available in NME and the inspector because there is no visual design implemented for these. Once ready it will be easy to hook this up, 

Testing:
I have tested on the GUI Editor. I have also tested on the inspector and NME to make sure that this code is not breaking those functionalities. 

https://github.com/BabylonJS/ThePirateCove/issues/11

### CODE REVIEW

File: sharedUiComponents/lines/color3LineComponent.tsx [hunk 1]
```diff
@@ -17,47 +17,99 @@ export interface IColor3LineComponentProps {
...
         if (props.isLinear) {
             this.state.color.toGammaSpaceToRef(this.state.color);
         }
 
         props.target._isLinearColor = props.isLinear; // so that replayRecorder can append toLinearSpace() as appropriate
+        this._colorPickerOpen = false;
+    }
+
+    private convertToColor3(color: string) {
+        if (color === "" || color === "transparent" || color === "Transparent") {
```
Comment (RaananW) [inline sharedUiComponents/lines/color3LineComponent.tsx:48]: you can use a regex or automatically convert the string to lowercase. `Transparent` is technically invalid.

File: sharedUiComponents/lines/color3LineComponent.tsx [hunk 2]
```diff
@@ -17,47 +17,99 @@ export interface IColor3LineComponentProps {
...
+            return new Color4(0, 0, 0, 0);
+        }
+
+        if (color.substring(0, 1) !== "#" || color.length !== 7) {
+            let d = document.createElement("div");
```
Comment (RaananW) [inline sharedUiComponents/lines/color3LineComponent.tsx:53]: all `let`s can be `const`s :-)

File: sharedUiComponents/lines/color3LineComponent.tsx [hunk 3]
```diff
@@ -17,47 +17,99 @@ export interface IColor3LineComponentProps {
...
+
+
+            let rgbArray = rgb.substring(4, rgb.length - 1)
+                .replace(/ /g, '')
+                .split(',');
+
+            return new Color3(parseInt(rgbArray[0]) / 255, parseInt(rgbArray[1]) / 255, parseInt(rgbArray[2]) / 255);
+        }
+
+        var r = parseInt(color.substring(1, 3), 16);
```
Comment (RaananW) [inline sharedUiComponents/lines/color3LineComponent.tsx:67]: consts.

File: sharedUiComponents/lines/color3LineComponent.tsx [hunk 4]
```diff
@@ -17,47 +17,99 @@ export interface IColor3LineComponentProps {
...
+        var g = parseInt(color.substring(3, 5), 16);
+        var b = parseInt(color.substring(5, 7), 16);
+
+        return Color3.FromInts(r, g, b);
```
Comment (RaananW) [inline sharedUiComponents/lines/color3LineComponent.tsx:71]: Color3 has `FromHexString` to do exactly what you are doing here (parsing a hex string)

File: sharedUiComponents/lines/color3LineComponent.tsx [hunk 5]
```diff
@@ -95,9 +151,9 @@ export class Color3LineComponent extends React.Component<IColor3LineComponentPro
 
         const store = this.state.color.clone();
         this.state.color.r = value;
-        this.setPropertyValue(this.state.color);
-        this.setState({ color: this.state.color });
-
+        let hex = this.state.color.toHexString();
```
Comment (RaananW) [inline sharedUiComponents/lines/color3LineComponent.tsx:154]: nit - const

**Review by RaananW — COMMENTED** [review]
A few changes, otherwise - LGTM!

File: sharedUiComponents/lines/color3LineComponent.tsx (hunk 1 above)
Comment (msDestiny14) [reply sharedUiComponents/lines/color3LineComponent.tsx:48]: I think I'm not going to even include it since you're right since "Transparent" is not valid.

**Comment by msDestiny14** [comment]
Looking to the import statements

**Review by RaananW — APPROVED** [review]

**Comment by msDestiny14** [comment]
Figured there would be some conflicts with the other PR. Let me clean those up. :)

**Pull request merged**